packages/tendermint-rpc/src/tendermint34/adaptor/responses.ts

```diff
@@ -252,6 +234,29 @@ function decodeConsensusParams(data: RpcConsensusParams): responses.ConsensusPar
   };
 }
 
+// for block results and genesis
+interface RpcValidatorUpdate {
+  /** hex encoded */
+  readonly address: string;
+  readonly pub_key:
+    | {
+        Sum: RpcPubkey;
+      }
+    | RpcPubkey;
```

Comment:
Nice solution, especially as https://github.com/tendermint/tendermint/issues/8583#issuecomment-1133301764 indicated this is a validator update specific issue.